scipy/interpolate/src/_dierckxmodule.cc

```diff
@@ -257,22 +592,54 @@ py_data_matrix(PyObject *self, PyObject *args)
     }
 
     try {
-        // heavy lifting happens here
-        fitpack::data_matrix(
-            static_cast<const double *>(PyArray_DATA(a_x)), m,
-            static_cast<const double *>(PyArray_DATA(a_t)), PyArray_DIM(a_t, 0),
-            k,
-            static_cast<const double *>(PyArray_DATA(a_w)),
-            extrapolate,
-            static_cast<double *>(PyArray_DATA(a_A)),     // output: (A, offset, nc)
-            static_cast<int64_t*>(PyArray_DATA(a_offs)),
-            &nc,
-            wrk.data()
-        );
-
-        // np.asarray(A), np.asarray(offset), int(nc)
-        PyObject *py_nc = PyLong_FromSsize_t(static_cast<Py_ssize_t>(nc));
-        return Py_BuildValue("(NNN)", PyArray_Return(a_A), PyArray_Return(a_offs), py_nc);
+        if( !periodic ) {
+            // heavy lifting happens here
+            fitpack::data_matrix(
+                static_cast<const double *>(PyArray_DATA(a_x)), m,
+                static_cast<const double *>(PyArray_DATA(a_t)), PyArray_DIM(a_t, 0),
+                k,
+                static_cast<const double *>(PyArray_DATA(a_w)),
+                extrapolate,
+                static_cast<double *>(PyArray_DATA(a_A)),     // output: (A, offset, nc)
+                static_cast<int64_t*>(PyArray_DATA(a_offs)),
+                &nc,
+                wrk.data()
+            );
+
+            // np.asarray(A), np.asarray(offset), int(nc)
+            PyObject *py_nc = PyLong_FromSsize_t(static_cast<Py_ssize_t>(nc));
+            return Py_BuildValue("(NNN)", PyArray_Return(a_A), PyArray_Return(a_offs), py_nc);
+        } else {
```

Comment:
Could you please split this wrapper into two, separate `data_matrix_periodic`. AFAIU, the C++ implementations are separate, the python usage is different, so there's no reason to not have two separate functions.